diff --git a/node_modules/react-native-callkeep/android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java b/node_modules/react-native-callkeep/android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java
index 025480a..6e37038 100644
--- a/node_modules/react-native-callkeep/android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java
+++ b/node_modules/react-native-callkeep/android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java
@@ -427,19 +427,19 @@ public class RNCallKeepModule extends ReactContextBaseJavaModule implements Life
         VoiceConnectionService.setPhoneAccountHandle(handle);
     }
 
-    @ReactMethod
+    //@ReactMethod
     public void unregisterEvents() {
         Log.d(TAG, "[RNCallKeepModule] unregisterEvents");
 
         this.hasListeners = false;
     }
 
-    @ReactMethod
+    //@ReactMethod
     public void displayIncomingCall(String uuid, String number, String callerName) {
         this.displayIncomingCall(uuid, number, callerName, false, null);
     }
 
-    @ReactMethod
+    //@ReactMethod
     public void displayIncomingCall(String uuid, String number, String callerName, boolean hasVideo) {
         this.displayIncomingCall(uuid, number, callerName, hasVideo, null);
     }
@@ -483,12 +483,12 @@ public class RNCallKeepModule extends ReactContextBaseJavaModule implements Life
         conn.onAnswer();
     }
 
-    @ReactMethod
+    //@ReactMethod
     public void startCall(String uuid, String number, String callerName) {
         this.startCall(uuid, number, callerName, false, null);
     }
 
-    @ReactMethod
+    //@ReactMethod
     public void startCall(String uuid, String number, String callerName, boolean hasVideo) {
         this.startCall(uuid, number, callerName, hasVideo, null);
     }
@@ -1129,10 +1129,15 @@ public class RNCallKeepModule extends ReactContextBaseJavaModule implements Life
 
         PhoneAccount.Builder builder = new PhoneAccount.Builder(handle, appName);
         if (isSelfManaged()) {
-            builder.setCapabilities(PhoneAccount.CAPABILITY_SELF_MANAGED);
+            int capabilities = PhoneAccount.CAPABILITY_SELF_MANAGED;
+            // Required for Android 14+ to avoid SecurityException on registerPhoneAccount
+            capabilities |= PhoneAccount.CAPABILITY_SUPPORTS_TRANSACTIONAL_OPERATIONS;
+            builder.setCapabilities(capabilities);
         }
         else {
-            builder.setCapabilities(PhoneAccount.CAPABILITY_CALL_PROVIDER);
+            // Required for Android 14+ to avoid SecurityException on registerPhoneAccount
+            int capabilities = PhoneAccount.CAPABILITY_CALL_PROVIDER | PhoneAccount.CAPABILITY_SUPPORTS_TRANSACTIONAL_OPERATIONS;
+            builder.setCapabilities(capabilities);
         }
 
         if (_settings != null && _settings.hasKey("imageName")) {
